plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/SnapshotsTable.java

```diff
@@ -81,29 +87,39 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Conne
     {
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
 
+        TableScan tableScan = buildTableScan(icebergTable, SNAPSHOTS);
         TimeZoneKey timeZoneKey = session.getTimeZoneKey();
-        icebergTable.snapshots().forEach(snapshot -> {
-            pagesBuilder.beginRow();
-            pagesBuilder.appendTimestampTzMillis(snapshot.timestampMillis(), timeZoneKey);
-            pagesBuilder.appendBigint(snapshot.snapshotId());
-            if (checkNonNull(snapshot.parentId(), pagesBuilder)) {
-                pagesBuilder.appendBigint(snapshot.parentId());
-            }
-            if (checkNonNull(snapshot.operation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.operation());
-            }
-            if (checkNonNull(snapshot.manifestListLocation(), pagesBuilder)) {
-                pagesBuilder.appendVarchar(snapshot.manifestListLocation());
-            }
-            if (checkNonNull(snapshot.summary(), pagesBuilder)) {
-                pagesBuilder.appendVarcharVarcharMap(snapshot.summary());
-            }
-            pagesBuilder.endRow();
+
+        tableScan.planFiles().forEach(fileScanTask -> {
+            DataTask dataTask = (DataTask) fileScanTask;
```

Comment:
Having to cast this here is kinda unfortunate, but the method that returns DataTasks is `private`. Looks safe though

Reply:
Let's at least add a `verify(fileScanTask.isDataTask(), ...)` before the cast and throw a useful error message

Reply:
Is 'asDataTask' possible here?